src/Symfony/Component/Lock/Lock.php

```diff
@@ -125,6 +147,7 @@ public function isAcquired()
     public function release()
     {
         $this->store->delete($this->key);
+        $this->dirty = false;
 
         if ($this->store->exists($this->key)) {
```

Comment:
unrelated, but just wondering: isn't this creating a race condition?

Reply:
No, unless you share the key with an other process (i'm talking about the key, not the resource).

The method `exists` check both the "resource" and the "owner". You may have a race condition by using pcntl_fork (or serializeing then share the key) and both processes try to release and acquire the lock on the same time. But IMO, that's a weird case.

BTW, we don't provide methods to check if the lock is acquired by someone else.